pkg/cmd/cli/cmd/rollback.go

```diff
@@ -0,0 +1,122 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	kubectl "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
+	kcmd "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd"
+
+	describe "github.com/openshift/origin/pkg/cmd/cli/describe"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+)
+
+const rollbackLongDesc = `
+Revert part of an application back to a previous deployment.
+
+When you run this command your deployment configuration will be updated to match
+the provided deployment. By default only the pod and container configuration
+will be changed and scaling or trigger settings will be left as-is. Note that
+environment variables and volumes are included in rollbacks, so if you've
+recently updated security credentials in your environment your previous
+deployment may not have the correct values.
+
+If you would like to review the outcome of the rollback, pass '--dry-run' to print
+a human-readable representation of the updated deployment configuration instead of
+executing the rollback. This is useful if you're not quite sure what the outcome
+will be.
+
+Examples:
+  Perform a rollback:
+
+  $ %[1]s %[2]s deployment-2
+
+  See what the rollback will look like, but don't perform the rollback:
+
+  $ %[1]s %[2]s deployment-1 --dry-run
+
+  Perform the rollback manually by piping the JSON of the new config back to %[1]s:
+
+  $ %[1]s %[2]s deployment-1 --output=json | %[1]s update deploymentConfigs deployment -f -
+`
+
+func NewCmdRollback(parentName string, name string, f *Factory, out io.Writer) *cobra.Command {
+	rollback := &deployapi.DeploymentConfigRollback{
+		Spec: deployapi.DeploymentConfigRollbackSpec{
+			IncludeTemplate: true,
+		},
+	}
+
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("%s <from-deployment>", name),
+		Short: "Rolls back a deploymentConfig to match a previous deployment.",
```

Comment:
Our docs should never say "deploymentConfig" - always deployment of deployment configuration (or part of an application).

This should match the longdesc